helix-term/src/commands.rs

```diff
@@ -4543,13 +4543,20 @@ fn shell_keep_pipe(cx: &mut Context) {
     );
 }
 
-fn shell_impl(
+fn shell_impl(shell: &[String], cmd: &str, input: Option<Rope>) -> anyhow::Result<(Tendril, bool)> {
+    tokio::task::block_in_place(|| {
+        let runtime = tokio::runtime::Handle::current();
+        runtime.block_on(shell_impl_async(shell, cmd, input))
+    })
```

Comment:
```suggestion
    tokio::task::block_in_place(|| helix_lsp::block_on(shell_impl_async(shell, cmd, input)))
```

This will become more colloquial in the code-base once #2267 is in